src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -433,19 +433,36 @@ _public_ int sd_is_mq(int fd, const char *path) {
         return 1;
 }
 
+static int vsock_bind_privileged_port(int fd) {
+        SocketAddress bind_address;
+        int r;
+
+        assert(fd >= 0);
+
+        r = socket_address_parse_vsock(&bind_address, "vsock::1023");
+        if (r < 0)
+                return r;
```

Comment:
hmm, why not just:

```c
union sockaddr_union sa = {
        .vm.svm_family = AF_VSOCK,
        .vm.svm_port = 1023,
}; 

for (;;) {
        r = RET_NERRNO(bind(fd, &sa.sa, sizeof(sa.vm));
        if (r != -EADDRINUSE || sa.vm.svm_port == 0)
                return r;
        sa.vm_svm_port--;
}
```

i.e. you don't need a whole SocketAddress here, the simple sockaddr_union suffices.